### ISSUE TITLE
Does not handle incorrect HTTP method errors well

### ISSUE THREAD

**Issue by jaredm4** [facebook/facebook-php-sdk-v4#369]
When I was using 'get' (instead of 'GET'), I would get the error back "Application does not have permission" for an page's albums. In other requests, I may just get back NULL.

My preferred fix is to have the SDK upper-case the HTTP method.

Though another appropriate fix would be to validate the HTTP methods being passed in and throw an appropriate error.
```
<?php

use Facebook\FacebookSession;
use Facebook\FacebookRequest;

require __DIR__ . '/vendor/autoload.php';

FacebookSession::setDefaultApplication('{{app id}}', '{{app secret}}');

$facebook_session = new FacebookSession('{{valid access token}}');

/** @var \Facebook\GraphAlbum $albums */
$albums = (new FacebookRequest(
    $facebook_session, 'get', '/{{page id}}/albums'
))->execute()->getGraphObjectList('Facebook\GraphAlbum');

var_dump($albums);
```
Output:
```Fatal error: Uncaught exception 'Facebook\FacebookPermissionException' with message '(#10) Application does not have permission for this action'```

**Comment by yguedidi** [comment]
The request is sent, the issue comes from your access token, it's not able to access that page album.
https://developers.facebook.com/docs/graph-api/reference/v2.2/page/albums#readperms

**Comment by jaredm4** [comment]
No, the access token is fine. Merely changing it from 'get' to 'GET' works. This has nothing to do with the token, but an incorrectly cased HTTP method (on my end). It shouldn't get far enough to make a request and provide an incorrect error.

**Comment by SammyK** [comment]
Glad you figured it out @jaredm4!

This can be closed as "won't fix" since `get` is not a valid HTTP verb.

**Comment by jaredm4** [comment]
Wait, no, that's not what I meant.  The purpose of this issue was to try to get HTTP method validation added, if we're not allowed to specify case-insensitive methods.

**Comment by SammyK** [comment]
Yeah, it wouldn't hurt to have some validation like in the [master branch](https://github.com/facebook/facebook-php-sdk-v4/blob/master/src/Facebook/FacebookRequest.php#L240).

**Comment by jaredm4** [comment]
That class in master also has a ```strtoupper()``` going on. Is this code slated for the next 4.0.x release? If so, we can surely close this ticket.

**Comment by gfosco** [comment]
Feel like submitting a PR against 4.0-dev @jaredm4 ?

**Comment by jaredm4** [comment]
I can take a look, sure. Currently racing through all our code updating old code to use 4.0 (v2.2 api) due to upcoming deadline. When done I can take a stab at it.

**Comment by SammyK** [comment]
It's slated for 4.1 which is a complete refactor of the SDK. It will be released pretty soon. :)

**Comment by roman02** [comment]
@SammyK What's "pretty soon"? In March? I've been postponing switching to API v2 until SDK 4.1.

**Comment by gfosco** [comment]
4.1 is just about complete, if not already launch-worthy.  I have to get the documentation loaded to the developers site before we can green-light it.

**Comment by SammyK** [comment]
@roman02 Just FYI, I posted an [Upgrade Guide: Facebook PHP SDK v3.x to v4.1](https://www.sammyk.me/upgrading-the-facebook-php-sdk-from-v3-x-to-v4-1). Hope that helps! :)
